astropy/cosmology/funcs.py

```diff
@@ -14,7 +21,41 @@
 __doctest_requires__ = {'*': ['scipy']}
 
 
-def z_at_value(func, fval, zmin=1e-8, zmax=1000, ztol=1e-8, maxfun=500):
+def _z_at_array(func, fvals, zmin, zmax, nbins=10000,
+                logspace=True, interpolation=None):
+    """Helper function to interpolate (func, z) over a grid for array input"""
+    if logspace:
+        zgrid = np.logspace(np.log10(zmin), np.log10(zmax), nbins)
+    else:
+        zgrid = np.linspace(zmin, zmax, nbins)
+
+    fgrid = func(zgrid)
+
+    fvals_val = fvals.value
+    fgrid_val = fgrid.to_value(fvals.unit)
+
+    if interpolation is None:
+        if USE_SCIPY:
+            interpolation = 'cubic'
+        else:
+            warnings.warn("""\
+SciPy not found, so falling back to linear numpy interpolation scheme
+for array z_at_value, instead of the normal cubic spline.""")
+            interpolation = 'linear'
+
+    assert interpolation in ['cubic', 'linear']
```

Comment:
This check could be integrated into the `if`-block by 

```
if interpolation == "cubic":
   ...
elif interpolation == "linear":
    ...
else:
    raise ValueError("[informative message here]")
```

Reply:
This actually is a debugging statement so I think it's good that optimization removes it, right?

Reply:
It's more that without the assertion, the function never checks for "linear". In essence the actual options for "interpolation" are `None`, `"cubic"` and *anything else*., eg. `42`. Were future additions to the code to add another interpolation scheme, "linear" would be checked with an "elif". This  would technically be an API change, since the code never checks for "linear".

Reply:
Drive-by comment: Please don't use `assert` unless it is a test code.

Reply:
Oh, sorry, I see what you meant! Okay, I am pushing a fix.